docs/doc/60-contributing/03-rfcs/20221013-iceberg-external-table.md

```diff
@@ -0,0 +1,225 @@
+
+### external-location
+
+No matter where the Iceberg is, like S3, GCS or OSS, if Databend support the storage, then the Iceberg should be accessable. Users need to tell databend where the Iceberg storage is and how should databend access the storage.
+
+### Type convention
+
+| Iceberg         | Note                                                     | Databend                             |
+| --------------- | -------------------------------------------------------- | ------------------------------------ |
+| `boolean`       | True or false                                            | `BOOLEAN`                            |
+| `int`           | 32 bit signed integer                                    | `INT32`                              |
+| `long`          | 64 bit signed integer                                    | `INT64`                              |
+| `float`         | 32 bit IEEE 754 float point number                       | `FLOAT`                              |
+| `double`        | 64 bit IEEE 754 float point number                       | `DOUBLE`                             |
+| `decimal(P, S)` | Fixed point decimal; precision P, scale S                | not supported                        |
+| `date`          | Calendar date without timezone or time                   | `DATE`                               |
+| `time`          | Timestamp without date, timezone                         | `TIMESTAMP`, convert date to today   |
+| `timestamp`     | Timestamp without timezone                               | `TIMESTAMP`, convert timezone to GMT |
+| `timestamptz`   | Timestamp with timezone                                  | `TIMESTAMP`                          |
+| `string`        | UTF-8 string                                             | `VARCHAR`                            |
+| `uuid`          | 16-byte fixed byte array, universally unique identifiers | `VARCHAR`                            |
+| `fixed(L)`      | Fixed-length byte array of length L                      | `VARCHAR`                            |
+| `binary`        | Arbitrary-length byte array                              | `VARCHAR`                            |
+| `struct`        | a tuple of typed values                                  | `OBJECT`                             |
+| `list`          | a collection of values with some element type            | `ARRAY`                              |
+| `map`           |                                                          | `OBJECT`                             |
+
+## Drawbacks
+
+None
+
+## Rationale and alternatives
+
+### Create Table from Iceberg Snapshots
+
+Iceberg provides a snapshot capability and tables can be created from them.
+
+```sql
+CREATE TABLE [IF NOT EXISTS] [db.]table_name
+(
+    <column_name> <data_type> [ NOT NULL | NULL] [ { DEFAULT <expr> }],
```

Comment:
If the iceberg table have already existed, do we still need to address the columns inside the external table? 

can we just create the external table with connection info ?

like:

```
CREATE EXTERNAL TABLE tbl ENGINE=iceberg ICEBERG_SNAPSHOT_LOCATION=(...
```

Reply:
This is a good idea and we can define tables like:

```sql
CREATE EXTERNAL TABLE tbl ENGINE=ICEBERG
ENGINE_OPTIONS=(...)
```

Reply:
I think we can support this first: always read table schema from the iceberg, so we don't need to handle the schema mismatch issues.

Reply:
The iceberg table engine is read-only? If so, we can define a new catalog: `ICEBERG` same as the hive table that we have already done. cc @ClSlaid

Reply:
We shall implement it step by step, for now, it's planned to be read-only. Schema evolution maybe supported in the future.

Reply:
By adding catalog support, we don't need to create tables anymore.

```sq
CREATE CATALOG my_iceberg TYPE='iceberg';
USE my_iceberg.test_db;
SELECT * from test_table;
```